src/app/views/query-response/pivot-items/pivot-items.tsx

```diff
@@ -84,6 +90,29 @@ export const getPivotItems = (properties: any) => {
   return pivotItems;
 };
 
+export const onPivotItemClick = (query: IQuery, item?: PivotItem) => {
+  if (!item) { return; }
+  const key = item.props.itemKey;
+  if (key) { trackTabClickEvent(query, key); }
+};
+
+function trackTabClickEvent(query: IQuery, tabKey: string) {
+  const queryUrl = query.selectedVerb + ' ' + query.sampleUrl;
+  switch (tabKey) {
+    case 'adaptive-cards': {
+      telemetry.trackEvent(TAB_CLICK_EVENT, { ComponentName: 'Adaptive Cards Tab', QueryUrl: queryUrl });
+      break;
+    }
+    case 'code-snippets': {
+      telemetry.trackEvent(TAB_CLICK_EVENT, { ComponentName: 'Code Snippets Tab', QueryUrl: queryUrl});
+      break;
+    }
+    default: {
+      break;
+    }
```

Comment:
To reduce code repetition of the trackEvent function, the switch can be changed so that it receives the only thing that is changing. in this case it is the component name

```suggestion
  let componentName = '';
  switch (tabKey) {
    case 'adaptive-cards': {
      componentName = 'Adaptive Cards Tab';
      break;
    }
    case 'code-snippets': {
      componentName = 'Code Snippets Tab';
      break;
    }
    default: {
      break;
    }
  }
  telemetry.trackEvent(TAB_CLICK_EVENT, { ComponentName: componentName, QueryUrl: queryUrl });
```

Another option, would be to pass the tabKey directly in the function and not do a switch case.

```suggestion
   telemetry.trackEvent(TAB_CLICK_EVENT, { ComponentName: tabKey, QueryUrl: queryUrl });
```

Reply:
I'll take the first suggestion, thanks.